### PR TITLE
Stripe integration + Refactor for Nav Bar

### PR DESCRIPTION
**Pull request by jassalss**
This is my PR for stripe integration but contains a massive refactor for Nav Bar. I had to change the Nav bar code because I had to fit the upgrade (payments) screen. Now NavBar is present on all screens except screens related to authentication and initial setup. Please check the pictures for more info. Links on Nav Bar are working now.

**With NavBar**

WithNavBar

**Without NavBar**

withoutNavBar

Thanks

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -28,8 +29,18 @@ const theme = createMuiTheme({
...
 });
 function App() {
+  let location = window.location;
+  let DisplayNavBar = location.pathname !== "/" &&
+    location.pathname !== "/auth-with-google" &&
+    location.pathname !== "/sign-up" &&
+    location.pathname !== "/login-in" &&
+    location.pathname !== "/profile_settings" &&
+    location.pathname !== "/availability" &&
+    location.pathname !== "/confirm" && <NavBar />;
```
Comment (sina-jamshidi) [inline client/src/App.js:39]: For maintainability and readability I think we can declare something like `displayNavBarPaths = ["/auth-with-google", so on...]`.
Then we can have `let DisplayNavBar = displayNavBarPaths.includes(location.pathname) && <NavBar />`

File: client/src/pages/Upgrade.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  },
+  greenColor: {
+    color: "#2e7d32",
+  },
+}));
+const Upgrade = () => {
+  const classes = useStyles();
+  const handleOnclick = async (planType) => {
+    const result = await fetch(
+      `http://localhost:5000/subscribe/5f770e13b2952ffe5546adff/${planType}`
```
Comment (sina-jamshidi) [inline client/src/pages/Upgrade.js:21]: should change that userId to be dynamic

File: server/api/stripe_confirm_webhook.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        session = event['data']['object']
+        if session["payment_status"] == "paid":
+            user_id = session["metadata"]["user_id"]
+            data_to_update = {"subscription_id": session["subscription"],
+                              "customer_id": session["customer"]}
+            query = {"_id": ObjectId(user_id)}
+            newValues = {"$set": data_to_update}
+            User.update(query, newValues)
+
+    return {}
```
Comment (sina-jamshidi) [inline server/api/stripe_confirm_webhook.py:44]: Nice!

**Review by sina-jamshidi — APPROVED** [review]
Couple of small changes otherwise okay to merge!

File: client/src/App.js (hunk 1 above)
Comment (jassalss) [reply client/src/App.js:39]: fixed.

File: client/src/pages/Upgrade.js (hunk 2 above)
Comment (jassalss) [reply client/src/pages/Upgrade.js:21]: fixed

**Pull request merged**